src/main/kotlin/com/autonomousapps/tasks/AdviceTask.kt

```diff
@@ -132,58 +137,85 @@ abstract class AdviceTask : DefaultTask() {
 
     val computedAdvice = advisor.compute(filterSpecBuilder())
     val advices = computedAdvice.getAdvices()
-    val advicePrinter = AdvicePrinter(computedAdvice)
-
-    var didGiveAdvice = false
-    val consoleReport = StringBuilder()
+    val consoleReport = ConsoleReport.build(computedAdvice)
 
-    if (!computedAdvice.filterRemove) {
-      advicePrinter.getRemoveAdvice()?.let {
-        consoleReport.add("Unused dependencies which should be removed:\n$it\n")
-        didGiveAdvice = true
-      }
-    }
+    adviceFile.writeText(advices.toJson())
+    advicePrettyFile.writeText(advices.toPrettyString())
+    adviceConsoleFile.writeText(consoleReport.toJson())
+    adviceConsolePrettyFile.writeText(consoleReport.toPrettyString())
 
-    if (!computedAdvice.filterAdd) {
-      advicePrinter.getAddAdvice()?.let {
-        consoleReport.add("Transitively used dependencies that should " +
-          "be declared directly as indicated:\n$it\n")
-        didGiveAdvice = true
-      }
-    }
+    chatter.chat("See machine-readable report at ${adviceFile.path}")
+    chatter.chat("See pretty report at ${advicePrettyFile.path}")
+    chatter.chat("See machine-readable console report at ${adviceConsoleFile.path}")
+    chatter.chat("See pretty console report at ${adviceConsolePrettyFile.path}")
 
-    if (!computedAdvice.filterChange) {
-      advicePrinter.getChangeAdvice()?.let {
-        consoleReport.add("Existing dependencies which should be modified " +
-          "to be as indicated:\n$it\n")
-        didGiveAdvice = true
-      }
+    if (consoleReport.isEmpty()) {
+      chatter.chat("Looking good! No changes needed")
+    } else {
+      chatter.chat("See console report at ${adviceConsoleFile.path}")
     }
+  }
 
-    if (!computedAdvice.filterCompileOnly) {
-      advicePrinter.getCompileOnlyAdvice()?.let {
-        consoleReport.add("Dependencies which could be compile-only:\n$it\n")
-        didGiveAdvice = true
+  internal data class ConsoleReport(
+    val addToApiAdvice: Set<Advice> = emptySet(),
+    val addToImplAdvice: Set<Advice> = emptySet(),
+    val removeAdvice: Set<Advice> = emptySet(),
+    val changeToApiAdvice: Set<Advice> = emptySet(),
+    val changeToImplAdvice: Set<Advice> = emptySet(),
+    val compileOnlyDependencies: Set<Dependency> = emptySet(),
+    val unusedProcsAdvice: Set<Advice> = emptySet()) {
+
+    fun isEmpty() = addToApiAdvice.isEmpty() &&
+      removeAdvice.isEmpty() &&
+      changeToApiAdvice.isEmpty() &&
+      changeToImplAdvice.isEmpty() &&
+      compileOnlyDependencies.isEmpty() &&
+      unusedProcsAdvice.isEmpty()
+
+    companion object {
+      fun build(computedAdvice: ComputedAdvice): ConsoleReport {
```

Comment:
There might be a better way to do this in kotlin. A builder ? My kotlin fluency...

Reply:
I think the method should be named `from`, not `build`. You're basically converting an instance of `ComputedAdvice` into an instance of `ConsoleReport`, and I think the right verb for that is "from".

I also found it surprising to see a bunch of `var`s. My instinct with Kotlin code is to use `val`s to a great extremity :) Having said that, to use vals instead of vars here, these would need to be mutable sets and you'd use `addAll()`, and there would be efficiency implications. Having said _that_, I don't know that this area here is an actual hotspot, so it may not matter.

If the choice is between `var` or a mutable set, I prefer the `var`. We can imagine using val and `by lazy { ... }`, but that might be harder to read and comprehend, and I prefer easier comprehension.

Reply:
So, I just rename, right ?

Reply:
Yes. Sorry, I may be being too verbose, but I'm basically thinking aloud so we can have a kind of async conversation in case there is more to react to.

I can, uh, try to _reduce_ that in the future.